File: Program's_Contributed_By_Contributors/Java_Programs/Misc/Pattern.java
public class TrianglePattern {
    public static void main(String[] args) {

        /*
                   1
                 1   1
               1   2   1
             1   3   3   1
           1   4   6   4   1
         1   5  10  10   5   1
       1   6  15  20  15   6   1
     1   7  21  35  35  21   7   1
                                          */

        int n = 8;
        for (int i = 0; i < n; i++)
        {
            int number = 1;
            System.out.printf("%" + (n - i) * 2 + "s", "");
            for (int j = 0; j <= i; j++)
            {
                System.out.printf("%4d", number);
                number = number * (i - j) / (j + 1);
            }
            System.out.println();
        }
    }
}

File: pattern.py
# Python Program to print star pattern 2
'''

* * * * *
* * * *
* * *
* *
*

'''
# define a function to print star shape
def star_pattern2(n):
    # Use outer for loop for rows

    for i in range(n, 0, -1 ):

        # Use inner for loop for columns / stars

        for j in range(1, i + 1):
            # print stars in each row
            print("* ", end="")

        # End of line after each row
        print()

n = int(input("Enter number of rows:"))
star_pattern2(n)
